File: VB/PivotGridCustomSummaryExample/CustomSummaryHelper.vb
Imports DevExpress.XtraPivotGrid
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace PivotGridCustomSummaryExample
	Friend NotInheritable Class CustomSummaryHelper

		Private Sub New()
		End Sub

		Public Shared Function GetGroupSummary(ByVal groupedDataSource As IEnumerable(Of IGrouping(Of Object, PivotDrillDownDataRow)), ByVal target As Object, ByVal dataFieldName As String) As Decimal
			Dim targetGroup = groupedDataSource.FirstOrDefault(Function(g) Object.Equals(g.Key, target))
			If targetGroup Is Nothing Then
				Return 0
			End If
			Return targetGroup.Sum(Function(r) Convert.ToDecimal(r(dataFieldName)))
		End Function

		Public Shared Function ShouldCalculateCustomValue(ByVal groupField As PivotGridField, ByVal e As PivotGridCustomSummaryEventArgs) As Boolean
			If Not groupField.Visible OrElse groupField.Area = PivotArea.FilterArea OrElse groupField.Area = PivotArea.DataArea Then
				Return True
			End If
			If groupField.Area = PivotArea.RowArea AndAlso (e.RowField Is Nothing OrElse e.RowField.AreaIndex < groupField.AreaIndex) Then
				Return True
			End If
			If groupField.Area = PivotArea.ColumnArea AndAlso (e.ColumnField Is Nothing OrElse e.ColumnField.AreaIndex < groupField.AreaIndex) Then
				Return True
			End If
			Return False
		End Function
	End Class
End Namespace

File: CS/PivotGridCustomSummaryExample/CustomSummaryHelper.cs
using DevExpress.XtraPivotGrid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PivotGridCustomSummaryExample
{
    static class CustomSummaryHelper
    {
        public static decimal GetGroupSummary(IEnumerable<IGrouping<object, PivotDrillDownDataRow>> groupedDataSource,
    object target, string dataFieldName)
        {
            var targetGroup = groupedDataSource.FirstOrDefault(g => object.Equals(g.Key, target));
            if (targetGroup == null)
                return 0;
            return targetGroup.Sum(r => Convert.ToDecimal(r[dataFieldName]));
        }

        public static bool ShouldCalculateCustomValue(PivotGridField groupField, PivotGridCustomSummaryEventArgs e)
        {
            if (!groupField.Visible || groupField.Area == PivotArea.FilterArea || groupField.Area == PivotArea.DataArea)
                return true;
            if (groupField.Area == PivotArea.RowArea &&
                (e.RowField == null || e.RowField.AreaIndex < groupField.AreaIndex))
                return true;
            if (groupField.Area == PivotArea.ColumnArea &&
                (e.ColumnField == null || e.ColumnField.AreaIndex < groupField.AreaIndex))
                return true;
            return false;
        }
    }
}
